packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx

```diff
@@ -68,22 +67,38 @@ const Zone = ({
       contentType: getContentTypeForTracking(window.location.pathname),
     });
   };
-  const enterSelectMode = () => {
-    setSelectModalData({ isOpen: true });
-  };
 
   const closeEditModal = () => {
     setEditModalData({ isOpen: false });
   };
-  const closeSelectModal = () => setSelectModalData({ isOpen: false });
 
   const onSelectFieldType = (widgetTypeName) => {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
     setNewFieldData({ widgetTypeName, fields });
-    setSelectModalData({ isOpen: false });
   };
 
   const onCancelNewField = () => setNewFieldData(null);
+  const addFieldDropdown = (
+    <AddFieldDropdown
+      disabled={newFieldData !== null}
+      onSelectField={onSelectFieldType}
+      fields={
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        widgetsArrayWithCondUid.filter(Boolean).map((widget) => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+          const { TYPE_NAME, CUSTOM_NAME } = widget;
+          return allFields.find(
+            (f) =>
+              f.type === TYPE_NAME &&
+              (CUSTOM_NAME === undefined || f.variant === CUSTOM_NAME),
+          );
+        })
+      }
+      triggerDataTestId={
+        isRepeatable === true ? "add-field-in-items" : undefined
```

Comment:
💡 
```suggestion
        isRepeatable ? "add-field-in-items" : undefined
```

Reply:
In SM we have an eslint rule that prevent that... --"

`An explicit comparison or type cast is required.eslint[@typescript-eslint/strict-boolean-expressions](https://typescript-eslint.io/rules/strict-boolean-expressions)`

Reply:
Maybe we could just remove it haha
We will need to review our setup in SM, lot and lot of things to improve there!

Reply:
Done, yeah a bit better